lib/tableau/extensions/post_extension.ex

```diff
@@ -129,6 +132,14 @@ defmodule Tableau.PostExtension do
           Enum.reject(posts, fn {post, _} -> DateTime.after?(post.date, DateTime.utc_now()) end)
         end
       end)
+      |> then(fn posts ->
+        if config.drafts do
+          Enum.reject(posts, fn {post, _} -> post.file =~ config.drafts end)
+        else
+          posts
+        end
+      end)
+      |> Enum.reject(fn {post, _} -> Map.get(post, :draft, false) == true end)
```

Comment:
I believe we want both of these to be guarded by the `if config.drafts do` condition, or else the config value doesn't control frontmatter, as well as it takes 2 passes on the posts

Reply:
So, in the end, lets actually combine all these `Enum.reject` calls into one, and inside the callback can determine if the post should be rejected based on all the conditions

Reply:
I guess I misunderstood the original goal: I thought that `config.drafts` set a folder which defaults to `/_drafts` that when posts are placed within, they don't get published.

I thought having `draft: true` in the frontmatter did the same, regardless of location.

I'm used to a similar behavior in jekyll or middleman, so that's where my head was at. I'm trying to imagine the use case of setting `config.drafts` to a folder, then not using the folder and only using the frontmatter... which you can do, but it feels odd.

Reducing the loops on `posts` does make sense though, I'll rewrite the current logic into one loop while we discuss.

Reply:
Logic does not look right, but tests pass… so now I have two problems.

- I think there needs to be an else there at least
- I need to check the tests to see why both pass with this configuration.

(Leaving this for myself later on my phone)

Reply:
Fixed the logic, although I feel it's a bit ugly.

Reply:
Oh, yes sorry I think I confused myself.

There needs to be a config value to render drafts at all, as well as a drafts_dir config value that can change which folder drafts are located in.

Your code (I haven't read the latest commits yet) only accounts for the latter kind of config value

Reply:
Ok, I'm writing tests and I'm slightly confused again @mhanberg. Since posts in `config.dir` are the only ones to be built, if you put a post in `_drafts` (which is what we were thinking about `config.drafts_dir`) would not be built. Are you imagining that config.drafts_dir lives *inside* `config.dir`? (ala `/posts/_drafts`?) If so, I'm working on it.

Reply:
I actually kind of think that a folder nested in `config.dir` might not be searched for posts either... but I'm going to have to set this aside for a moment.